lib/yargs-parser.ts

```diff
@@ -225,7 +225,7 @@ export class YargsParser {
       if (arg !== '--' && isUnknownOptionAsArg(arg)) {
         pushPositional(arg)
       // ---, ---=, ----, etc,
-      } else if (truncatedArg.match(/---+(=|$)/)) {
+      } else if (truncatedArg.match(/^---+(=|$)/)) {
```

Comment:
bother you to add a test for this?

Reply:
https://github.com/yargs/yargs-parser/pull/434/commits/94eeb8d76e210cdb445e6f856719c3d953ffca76 🙈

Reply:
🔔 @bcoe